src/vc_util/src/custom.rs

```diff
@@ -0,0 +1,189 @@
+// Custom utils for handling specific VCs/VPs.
+
+use crate::issuer_api::{ArgumentValue, CredentialSpec};
+use crate::{
+    inconsistent_jwt_claims, validate_claim, validate_claims_match_spec,
+    verify_ii_presentation_jwt_with_canister_ids, CredentialVerificationError,
+    PresentationVerificationError, VcFlowSigners,
+};
+use candid::Principal;
+use identity_credential::validator::JwtValidationError;
+use std::collections::HashMap;
+
+/// Validates the provided presentation `vp_jwt`, checking the following conditions:
+///  - the presentation is a valid presentation of the IC Attribute Sharing flow;
+///    in particular the presentation contains IdAlias and a matching VerifiedAdult credentials
+///  - both credentials are valid wrt. `vc_flow_signers`, `root_pk_raw`, and `current_time_ns`
+///  - `effective_vc_subject` matches the principal to which the presentation applies
+pub fn validate_verified_adult_presentation(
+    vp_jwt: &str,
+    effective_vc_subject: Principal,
```

Comment:
What is the effective vc subject?

Reply:
Maybe a comment explaining this in more detail might be nice?

Reply:
Done.